### PR TITLE
Reorder kryo.register statements in ADAMKryoRegistrator

### LINKED ISSUE #1146: Reorder kryo.register statements in ADAMKryoRegistrator
I'd like to clean up `ADAMKyroRegistrator` by removing package imports, using fully qualified class names, and sorting alphabetically, with blank lines between packages.  E.g.

```scala
// package org.bdgenomics.adam.models
kryo.register(classOf[org.bdgenomics.adam.models.Coverage], new AvroSerializer[org.bdgenomics.adam.models.Coverage])
kryo.register(classOf[org.bdgenomics.adam.models.ReferenceRegion], new AvroSerializer[org.bdgenomics.adam.models.ReferenceRegion])

// package org.bdgenomics.formats.avro
kryo.register(classOf[org.bdgenomics.formats.avro.AlignmentRecord], new AvroSerializer[org.bdgenomics.formats.avro.AlignmentRecord])
kryo.register(classOf[org.bdgenomics.formats.avro.Contig], new AvroSerializer[org.bdgenomics.formats.avro.Contig])

// package scala
kryo.register(classOf[scala.Array[org.bdgenomics.adam.models.Coverage]])
kryo.register(classOf[scala.Array[org.bdgenomics.adam.models.ReferenceRegion]])
kryo.register(classOf[scala.Array[org.bdgenomics.formats.avro.AlignmentRecord]])
kryo.register(classOf[scala.Array[org.bdgenomics.formats.avro.Contig]])
```

The way it is now is confusing and has resulted in missing and duplicated register statements.

### PR DESCRIPTION
**Pull request by heuermh**
Fixes #1146

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/serialization/ADAMKryoRegistrator.scala [hunk 1]
```diff
@@ -264,20 +112,139 @@ class ADAMKryoRegistrator extends KryoRegistrator {
...
+
+    // org.bdgenomics.adam.models
+    kryo.register(classOf[org.bdgenomics.adam.models.CDS])
+    kryo.register(classOf[org.bdgenomics.adam.models.Coverage])
+    kryo.register(classOf[org.bdgenomics.adam.models.Exon])
+    kryo.register(classOf[org.bdgenomics.adam.models.MultiContigNonoverlappingRegions])
+    kryo.register(classOf[org.bdgenomics.adam.models.NonoverlappingRegions])
+    kryo.register(classOf[org.bdgenomics.adam.models.RecordGroup])
+    kryo.register(classOf[org.bdgenomics.adam.models.RecordGroupDictionary])
+    kryo.register(classOf[org.bdgenomics.adam.models.ReferencePosition], new org.bdgenomics.adam.models.ReferencePositionSerializer)
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/serialization/ADAMKryoRegistrator.scala]: Nit: break long lines

File: adam-core/src/main/scala/org/bdgenomics/adam/serialization/ADAMKryoRegistrator.scala [hunk 2]
```diff
@@ -264,20 +112,139 @@ class ADAMKryoRegistrator extends KryoRegistrator {
...
+    kryo.register(classOf[org.bdgenomics.adam.models.ReferenceRegion])
+    kryo.register(classOf[org.bdgenomics.adam.models.ReferencePositionPair], new org.bdgenomics.adam.models.ReferencePositionPairSerializer)
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/serialization/ADAMKryoRegistrator.scala]: Nit: break long lines

File: adam-core/src/main/scala/org/bdgenomics/adam/serialization/ADAMKryoRegistrator.scala [hunk 3]
```diff
@@ -264,20 +112,139 @@ class ADAMKryoRegistrator extends KryoRegistrator {
...
+    kryo.register(classOf[org.bdgenomics.adam.models.SAMFileHeaderWritable])
+    kryo.register(classOf[org.bdgenomics.adam.models.SequenceDictionary])
+    kryo.register(classOf[org.bdgenomics.adam.models.SequenceRecord])
+    kryo.register(classOf[org.bdgenomics.adam.models.SingleReadBucket], new org.bdgenomics.adam.models.SingleReadBucketSerializer)
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/serialization/ADAMKryoRegistrator.scala]: Nit: break long lines

File: adam-core/src/main/scala/org/bdgenomics/adam/serialization/ADAMKryoRegistrator.scala [hunk 4]
```diff
@@ -264,20 +112,139 @@ class ADAMKryoRegistrator extends KryoRegistrator {
...
+
+    // org.bdgenomics.adam.rdd
+    kryo.register(classOf[org.bdgenomics.adam.rdd.GenomeBins])
+
+    // org.bdgenomics.adam.rdd.read
+    kryo.register(classOf[org.bdgenomics.adam.rdd.read.FlagStatMetrics])
+    kryo.register(classOf[org.bdgenomics.adam.rdd.read.DuplicateMetrics])
+
+    // org.bdgenomics.adam.rdd.read.realignment
+    kryo.register(classOf[org.bdgenomics.adam.rdd.read.realignment.IndelRealignmentTarget], new org.bdgenomics.adam.rdd.read.realignment.IndelRealignmentTargetSerializer)
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/serialization/ADAMKryoRegistrator.scala]: Nit: break long lines for the following 3

File: adam-core/src/main/scala/org/bdgenomics/adam/serialization/ADAMKryoRegistrator.scala [hunk 5]
```diff
@@ -264,20 +112,139 @@ class ADAMKryoRegistrator extends KryoRegistrator {
...
+    kryo.register(classOf[org.bdgenomics.adam.rich.DecadentRead])
+    kryo.register(classOf[org.bdgenomics.adam.rich.ReferenceSequenceContext])
+    kryo.register(classOf[org.bdgenomics.adam.rich.RichAlignmentRecord])
+    kryo.register(classOf[org.bdgenomics.adam.rich.RichVariant])
+
+    // org.bdgenomics.adam.util
+    kryo.register(classOf[org.bdgenomics.adam.util.MdTag])
+    kryo.register(classOf[org.bdgenomics.adam.util.QualityScore])
+    kryo.register(classOf[org.bdgenomics.adam.util.ReferenceContigMap])
+    kryo.register(classOf[org.bdgenomics.adam.util.TwoBitFile], new org.bdgenomics.adam.util.TwoBitFileSerializer)
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/serialization/ADAMKryoRegistrator.scala]: Nit: break long lines

File: adam-core/src/main/scala/org/bdgenomics/adam/serialization/ADAMKryoRegistrator.scala [hunk 6]
```diff
@@ -264,20 +112,139 @@ class ADAMKryoRegistrator extends KryoRegistrator {
...
+
+    // org.bdgenomics.formats.avro
+    kryo.register(classOf[org.bdgenomics.formats.avro.AlignmentRecord], new AvroSerializer[org.bdgenomics.formats.avro.AlignmentRecord])
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/serialization/ADAMKryoRegistrator.scala]: Nit: break long lines for all these lines

**Comment by fnothaft** [comment]
LGTM! Dropped a few nits about long lines. What do you think about pulling the `AvroSerializer` bits out into their own files?

**Comment by heuermh** [comment]
Dude, get a wider monitor!  ;)

**Comment by fnothaft** [comment]
Merged. Thanks @heuermh!

**Comment by heuermh** [comment]
Thank you!

**Pull request merged**